Packet++/header/DoIpLayer.h

```diff
@@ -0,0 +1,1802 @@
+		void setReservedIso(const std::array<uint8_t, DOIP_RESERVED_ISO_LEN>& reservedIso);
+
+		/// @brief Gets pointer to reserved OEM bytes if present.
+		const uint8_t* getReservedOem() const;
+
+		/// @brief Sets the reserved OEM bytes.
+		void setReservedOem(const std::array<uint8_t, DOIP_RESERVED_OEM_LEN>& reservedOem);
+
+		/// @brief Checks if OEM reserved bytes are present.
+		bool hasReservedOem() const;
+
+		/// @brief Clears the OEM reserved bytes.
+		void clearReservedOem();
+
+		/// @brief Returns a human-readable summary of the message.
+		std::string getSummary() const;
+
+		/// @brief Returns the DoIP payload type for this message.
+		DoIpPayloadTypes getPayloadType() const override
+		{
+			return DoIpPayloadTypes::ROUTING_ACTIVATION_REQUEST;
+		}
+
+		/// @brief Get a pointer to Routing activation request message
+		/// Notice this points directly to the data, so any change will modify
+		/// the actual packet data
+		/// @return A pointer to a routing_activation_request structure containing the data
+		routing_activation_request* getRoutingRequest() const
+		{
+			return reinterpret_cast<routing_activation_request*>(m_Data);
+		}
+
+	private:
+		bool _hasReservedOem;  ///< Whether OEM bytes are set.
+
+		static constexpr size_t ACTIVATION_TYPE_LEN = 1;
+		static constexpr size_t FIXED_LEN =
+		    DOIP_HEADER_LEN + DOIP_SOURCE_ADDRESS_LEN + ACTIVATION_TYPE_LEN + DOIP_RESERVED_ISO_LEN;
+		static constexpr size_t OPT_LEN = FIXED_LEN + DOIP_RESERVED_OEM_LEN;
+	};
+
+	//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|
+	// DoIpRoutingActivationResponse|
+	//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|
+
+	/// @class DoIpRoutingActivationResponse
+	/// @brief Represents a DoIP Routing Activation Response message.
+	///
+	/// Provides parsing and construction for Routing Activation Response messages
+	/// as defined by the DoIP protocol.
+	class DoIpRoutingActivationResponse : public DoIpLayer
+	{
+	public:
+		/// @brief Constructs the layer from raw DoIP packet data.
+		/// @param[in] data Pointer to raw packet data.
+		/// @param[in] dataLen Length of the raw data.
+		/// @param[in] prevLayer Pointer to the previous layer.
+		/// @param[in] packet Pointer to the parent packet instance.
+		DoIpRoutingActivationResponse(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		/// @brief Constructs the message from field values.
+		/// @param[in] logicalAddressExternalTester Logical address of the external tester.
+		/// @param[in] sourceAddress ECU source address.
+		/// @param[in] responseCode The routing response code.
+		DoIpRoutingActivationResponse(uint16_t logicalAddressExternalTester, uint16_t sourceAddress,
+		                              DoIpRoutingResponseCodes responseCode);
+
+/// @struct  routing_activation_response
+/// Represents the Routing Activation Response data in DoIP.
+/// Routing Activation Response message structure (extends DoIP header).
+#pragma pack(push, 1)
+		struct routing_activation_response : doiphdr
```

Comment:
Since we have getters and setters for all fields, maybe make the struct private?

Same for the other structs in the other classes below